ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/result-filter/index.tsx

```diff
@@ -0,0 +1,114 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import styled from '../../styles/styled-components'
+
+import { Button, buttonTypeEnum } from '../button'
+
+import MarionetteRegionContainer from '../../container/marionette-region-container'
+import * as FilterBuilderView from '../../../component/filter-builder/filter-builder.view'
+import * as FilterBuilderModel from '../../../component/filter-builder/filter-builder'
+
+type Props = {
+  removeFilter: () => void
+  saveFilter: (transformToCql: () => any) => void
+  hasFilter: Boolean
+  resultFilter?: {}
+  isList?: Boolean
+}
+
+const Root = styled<Props, 'div'>('div')`
+  display: block;
+  padding: ${props => props.theme.minimumSpacing};
+`
+
+const EditorFooter = styled<Props, 'div'>('div')`
+  padding-top: 30px;
+  white-space: nowrap;
+
+  button {
+    display: inline-block;
+    width: ${props => (props.hasFilter && '49%') || '100%'};
+    ${props => props.hasFilter && 'margin-right: 1%;'};
+  }
+
+  .footer-remove {
+    display: ${props => (props.hasFilter && 'inline-block') || 'none'};
+  }
+`
+
+class View extends React.Component<Props, { filterCqlCallback: () => any }> {
+  view: any
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      filterCqlCallback: () => {},
+    }
+  }
+
+  onViewBind = (view: any) => {
+    this.view = view
+
+    this.view.turnOnEditing()
+    this.view.turnOffNesting()
+    this.props.resultFilter && this.view.deserialize(this.props.resultFilter)
+
+    this.setState({
+      filterCqlCallback: this.view.transformToCql.bind(this.view),
+    })
+  }
+
+  onSave = () => {
+    this.props.saveFilter(this.state.filterCqlCallback())
+  }
+
+  render = () => {
+    return (
+      <Root
+        {...this.props}
+        className={`result-filter ${this.props.hasFilter && 'has-filter'} ${this
+          .props.isList && 'is-list'}`}
+      >
+        <div className="editor-properties">
+          <MarionetteRegionContainer
+            className="editor-properties"
+            view={FilterBuilderView}
+            bindView={this.onViewBind.bind(this)}
+            viewOptions={{
+              model: new FilterBuilderModel({ isResultFilter: true }),
+            }}
+          />
+        </div>
+        <EditorFooter {...this.props}>
+          <Button
+            className="footer-remove"
+            buttonType={buttonTypeEnum.negative}
+            text="Remove Filter"
+            onClick={this.props.removeFilter}
+          />
+          <Button
+            className="footer-save"
+            buttonType={buttonTypeEnum.positive}
+            text="Save Filter"
+            onClick={this.onSave.bind(this)}
```

Comment:
`onSave` is an arrow function and therefore shouldn't require you to `.bind(this)`. same for `onViewBind.bind(this)`